scrapper.py

```diff
@@ -23,75 +36,310 @@ class IncorrectNumberOfArticlesError(Exception):
+    """
+    Recursive Crawler implementation
+    """
+
+    def find_articles(self):
+        """
+        Finds articles
+        """
+        article_id = 1
+        while len(self.urls) < self.max_articles and self.seed_urls:
+            seed_url = self.seed_urls.pop()
+            try:
+                soup = self.process_page(seed_url, article_id)
+            except BadStatusCode:
+                continue
+            else:
+                self.seed_urls = self.get_search_urls(soup)
+                article_id += 1
+        self.urls = self.urls[:self.max_articles]
+
+    def get_search_urls(self, soup):
         """
         Returns seed_urls param
         """
-        pass
+        for idx, link in enumerate(soup.find_all('a')):
+            if link.get('href'):
+                if res := re.findall(r'https://www.zvezdaaltaya.ru/.+', link.get('href')):
+                    if res[0] not in self.seen_urls:
+                        self.seed_urls.append(res[0])
+                        self.seen_urls.add(res[0])
+            if idx % 10 == 0:
+                with open(constants.TO_PARSE_URLS, 'w', encoding='utf-8') as file:
+                    file.write('\n'.join(self.seed_urls))
+                with open(constants.SEEN_URLS, 'w', encoding='utf-8') as file:
+                    file.write('\n'.join(self.seen_urls))
+        return self.seed_urls
 
 
 class ArticleParser:
     """
     ArticleParser implementation
     """
     def __init__(self, full_url: str, article_id: int):
-        pass
+        self.article = Article(full_url, article_id)
 
     def _fill_article_with_text(self, article_soup):
-        pass
+        for paragraph in article_soup.find_all('p')[:-3]:
+            self.article.text += paragraph.text + ' '
 
     def _fill_article_with_meta_information(self, article_soup):
-        pass
+        self.article.title = re.sub(r' - Звезда Алтая', '', article_soup.find('title').text)
+        date_str = article_soup.find('span', {'class': 'mg-blog-date'}).text.strip()
+        self.article.date = self.unify_date_format(date_str)
+        self.article.author = article_soup.find('h4', {'class': 'media-heading'}).find('a').text
 
     @staticmethod
     def unify_date_format(date_str):
         """
         Unifies date format
         """
-        pass
+        month_dict = {
+            'Янв': 1,
+            'Фев': 2,
+            'Мар': 3,
+            'Апр': 4,
+            'Май': 5,
+            'Июн': 6,
+            'Июл': 7,
+            'Авг': 8,
+            'Сен': 9,
+            'Окт': 10,
+            'Ноя': 11,
+            'Дек': 12
+        }
+
+        month, day, year = date_str.split()
+        day = int(day[:-1])
+        year = (int(year))
+        month = month_dict[month]
+        return datetime(year, month, day)
 
     def parse(self):
         """
         Parses each article
         """
-        pass
+        article_page = Crawler.get_page(self.article.url)
```

Comment:
good intention, poor decision. Crawler and Article Parser should not know about each other. However, as there is the same thing to do, nice idea to have it in one place. Find better solution

Reply:
the same applies to everything, for example using the headers defined in Crawler

Reply:
thanks! made it independent